lnwallet/channel.go

```diff
@@ -6828,9 +6831,16 @@ func (lc *LightningChannel) MaxFeeRate(maxAllocation float64) chainfee.SatPerKWe
 	// Ensure the fee rate doesn't dip below the fee floor.
 	_, weight := lc.availableBalance()
 	maxFeeRate := maxFee / (float64(weight) / 1000)
-	return chainfee.SatPerKWeight(
+	feeRate := chainfee.SatPerKWeight(
 		math.Max(maxFeeRate, float64(chainfee.FeePerKwFloor)),
 	)
+
+	// Cap anchor fee rates.
+	if lc.channelState.ChanType.HasAnchors() && feeRate > maxAnchorFeeRate {
```

Comment:
above we use `commitType == lnwallet.CommitmentTypeAnchorsZeroFeeHtlcTx`, should that match here? otherwise the maximum will also apply to `anchor_outputs` channels?

Reply:
Yes, see this commit in the zero htlc fee PR: https://github.com/lightningnetwork/lnd/pull/4840/commits/7b0212f21aae6ce2d137359c4d0fa3a7fdb3c36d#diff-e308f199858d836b06a0c920d2dd019fbb3acb7ef0d9e3d8a360601623128d78R322 

So we'll always set both the anchors bit and the zero fee htlc bit if `ommitType == lnwallet.CommitmentTypeAnchorsZeroFeeHtlcTx`

Reply:
Correct that this'll apply to "regular" anchors as well, though with the dependent PR we essentially don't advertise it at all from teh perspective of the public network.

Reply:
Yeah, it will also apply the the existing few legacy anchor channels, but I think that's okay.